ctk/idp/coverage/NotTested.md

```diff
@@ -0,0 +1,59 @@
+| 3.3.4 - Processing Rules | If S2 includes one or more \<saml:SubjectConfirmation> elements, then S1 MUST include at least one \<SAML:SubjectConfirmation> element such that S1 can be confirmed in the manner described by at least one \<SAML:SubjectConfirmation> element in S2. | "Confirming" in \<SAML:SubjectConfirmation> is a very difficult task. For now only the \<SAML:SubjectConfirmation> Method attributes are compared, rather than actual confirmation.
+| 3.4.1 - Element <AuthnRequest> | If \[ForceAuthn is\] "true", the identity provider MUST authenticate the presenter directly rather than rely on a previous security context. If a value is not provided, the default is "false". | Hard to test because how they use “previous security context” is implementation-specific. We could catch most cases by handling cookies and/or session variables, for example, but it wouldn’t cover all cases.
+| 3.4.1 - Element <AuthnRequest> | If \[isPassive is\] "true", the identity provider and the user agent itself MUST NOT visibly take control of the user interface from the requester and interact with the presenter in a noticeable fashion. | Impossible to test because the tests cannot reliably tell if an HTML response is going to display anything in the browser. (for example, javascript could be embedded in the HTML that dynamically changes the HTML).
+| 3.4.1 - Element <AuthnRequest> | If both ForceAuthn and IsPassive are "true", the identity provider MUST NOT freshly authenticate the presenter unless the constraints of IsPassive can be met. | Impossible to test. See IsPassive.
+| 3.4.1 - Element <AuthnRequest> | AssertionConsumerServiceIndex usage | Can't be tested until a second binding (Artifact) is supported in the test kit, since we won't know if they're just relying on the default.
+| 3.4.1 - Element <AuthnRequest> | The identity provider MUST have a trusted means to map the \[AssertionConsumerServiceIndex\] index value in the attribute to a location associated with the requester. \[SAMLMeta\] provides one possible mechanism. | Hard to test because we don’t know what method the IdP is going to use.
+| 3.4.1 - Element <AuthnRequest> | \[AssertionConsumerServiceUrl\] Specifies by value the location to which the <Response> message MUST be returned to the requester.  | Can't be tested until a second binding (Artifact) is supported in the test kit, since we won't know if they're just relying on the default.
+| 3.4.1 - Element <AuthnRequest> | The responder MUST ensure by some means that the value specified is in fact associated with the requester. \[SAMLMeta\] provides one possible mechanism; signing the enclosing <AuthnRequest> message is another. | Hard to test because we don’t know what method the IdP is going to use.
```

Comment:
- `specified \[by the AssertionConsumerServiceURL\] is in fact`
- ❓ Not sure what you mean by `method`. We should be able to test this by providing an incorrect ACS URL especially since we only support metadata.